pandas/core/arrays/_reshaping.py

```diff
@@ -1,135 +1,13 @@
 """
 Utilities for implementing 2D compatibility for 1D ExtensionArrays.
 """
-from functools import wraps
 from typing import Tuple
 
 import numpy as np
 
 from pandas._libs.lib import is_integer
 
 
-def implement_2d(cls):
```

Comment:
Hmm sorry, this mixes a move and some changes.

I had to move it to `arrays/base.py` so that I could import ExtensionArray. Actually, 1 sec. I think we can just skip for EA.